src/DataBuilder.php

```diff
@@ -0,0 +1,803 @@
+            ->setCodeVersion($this->getCodeVersion())
+            ->setPlatform($this->getPlatform())
+            ->setLanguage($this->getLanguage())
+            ->setFramework($this->getFramework())
+            ->setContext($this->getContext())
+            ->setRequest($this->getRequest())
+            ->setPerson($this->getPerson())
+            ->setServer($this->getServer())
+            ->setCustom($this->getCustom($toLog, $context))
+            ->setFingerprint($this->getFingerprint())
+            ->setTitle($this->getTitle())
+            ->setUuid($this->getUuid())
+            ->setNotifier($this->getNotifier());
+        return $data;
+    }
+
+    protected function getEnvironment()
+    {
+        return $this->environment;
+    }
+
+    protected function getBody($toLog, $context)
+    {
+        $baseException = $this->getBaseException();
+        if ($toLog instanceof ErrorWrapper) {
+            $content = $this->getErrorTrace($toLog);
+        } elseif ($toLog instanceof $baseException) {
+            $content = $this->getExceptionTrace($baseException);
+        } else {
+            $scrubFields = $this->getScrubFields();
+            $content = $this->getMessage($toLog, self::scrub($context, $scrubFields));
+        }
+        return new Body($content);
+    }
+
+    protected function getErrorTrace(ErrorWrapper $error)
+    {
+        return $this->makeTrace($error, $error->getClassName());
+    }
+
+    /**
+     * @param \Throwable|\Exception $exc
+     * @return Trace|TraceChain
+     */
+    protected function getExceptionTrace($exc)
+    {
+        $chain = array();
+        $chain[] = $this->makeTrace($exc);
+
+        $previous = $exc->getPrevious();
+
+        $baseException = $this->getBaseException();
+        while ($previous instanceof $baseException) {
+            $chain[] = $this->makeTrace($previous);
+            $previous = $exc->getPrevious();
+        }
+
+        if (count($chain) > 1) {
+            return new TraceChain($chain);
+        }
+        return new Trace($chain[0], $chain[0]->getException());
+    }
+
+    /**
+     * @param \Throwable|\Exception $exception
+     * @param string $classOverride
+     * @return Trace
+     */
+    public function makeTrace($exception, $classOverride = null)
+    {
+        $frames = $this->makeFrames($exception);
+        $excInfo = new ExceptionInfo(
+            Utilities::coalesce($classOverride, get_class($exception)),
+            $exception->getMessage()
+        );
+        return new Trace($frames, $excInfo);
+    }
+
+    public function makeFrames($exception)
+    {
+        $frames = array();
+        foreach ($this->getTrace($exception) as $frameInfo) {
+            $filename = Utilities::coalesce($this->tryGet($frameInfo, 'file'), '<internal>');
+            $lineno = Utilities::coalesce($this->tryGet($frameInfo, 'line'), 0);
+            $method = $frameInfo['function'];
+            // TODO 4 (arguments are in $frame)
```

Comment:
At some point it would be good to add the arguments into the payload at this point we don't